model.py

```diff
@@ -1,466 +1,214 @@
-        instance.setVariableFloat("y", random.uniform(min_pos, max_pos))
-        instance.setVariableFloat("z", random.uniform(min_pos, max_pos))
-
-        # Generate a random velocity direction
-        fx = random.uniform(-1, 1)
-        fy = random.uniform(-1, 1)
-        fz = random.uniform(-1, 1)
-        # Generate a random speed between 0 and the maximum initial speed
-        fmagnitude = random.uniform(min_speed, max_speed)
-        # Use the random speed for the velocity.
-        vec3Normalize(fx, fy, fz)
-        vec3Mult(fx, fy, fz, fmagnitude)
-
-        # Set these for the agent.
-        instance.setVariableFloat("fx", fx)
-        instance.setVariableFloat("fy", fy)
-        instance.setVariableFloat("fz", fz)
-
-    cudaSimulation.setPopulationData(population)
-
-"""
-  Execution
-"""
-cudaSimulation.simulate()
-
-"""
-  Export Pop
-"""
-# cudaSimulation.exportData("end.xml")
-
-if pyflamegpu.VISUALISATION:
-    visualisation.join()
+    # If visualisation is enabled, use it.
+
+    if pyflamegpu.VISUALISATION:
+        visualisation = simulation.getVisualisation()
+        # Configure the visualiastion.
+        INIT_CAM = ENV_MAX * 1.25
+        visualisation.setInitialCameraLocation(INIT_CAM, INIT_CAM, INIT_CAM)
+        visualisation.setCameraSpeed(0.01)
+        vis_agent = visualisation.addAgent("Circle")
+        # Position vars are named x, y, z so they are used by default
+        # Set the model to use, and scale it.
+        vis_agent.setModel(pyflamegpu.ICOSPHERE)
+        vis_agent.setModelScale(1 / 10.0)
+        # Optionally render the Subdivision of spatial messaging
+        if VISUALISE_COMMUNICATION_GRID:
+            ENV_MIN = 0
+            DIM = int(math.ceil((ENV_MAX - ENV_MIN) / RADIUS))  # Spatial partitioning scales up to fit none exact environments
+            DIM_MAX = DIM * RADIUS
+            pen = visualisation.newLineSketch(1, 1, 1, 0.2)  # white
+            # X lines
+            for y in range(0, DIM + 1):
+                for z in range(0, DIM + 1):
+                    pen.addVertex(ENV_MIN, y * RADIUS, z * RADIUS)
+                    pen.addVertex(DIM_MAX, y * RADIUS, z * RADIUS)
+            # Y axis
+            for x in range(0, DIM + 1):
+                for z in range(0, DIM + 1):
+                    pen.addVertex(x * RADIUS, ENV_MIN, z * RADIUS)
+                    pen.addVertex(x * RADIUS, DIM_MAX, z * RADIUS)
+            # Z axis
+            for x in range(0, DIM + 1):
+                for y in range(0, DIM + 1):
+                    pen.addVertex(x * RADIUS, y * RADIUS, ENV_MIN)
+                    pen.addVertex(x * RADIUS, y * RADIUS, DIM_MAX)
+        # Activate the visualisation.
+        visualisation.activate()
+
+    # Initialise the simulation
+    simulation.initialise(sys.argv)
+
+    # Generate a population if an initial states file is not provided
+    if not simulation.getSimulationConfig().input_file:
```

Comment:
You removed this method, I will fix it.

Reply:
I must have dev'd this with an install from an older wheel. Well spotted.

Reply:
I was just testing two py/c examples for my dissertation meeting and found it, then saw it was present here too.